sdk/python/sawtooth_sdk/workload/workload_generator.py

```diff
@@ -73,64 +73,69 @@ def run(self):
 
     @asyncio.coroutine
     def _simulator_loop(self):
+        while True:
+            yield from asyncio.sleep(self._rate)
+            with self._lock:
```

Comment:
The workload generator uses 1 coroutine and threads. As such asyncio.Lock should not be used in threads.